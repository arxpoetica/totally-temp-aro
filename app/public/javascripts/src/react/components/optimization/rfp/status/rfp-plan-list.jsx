import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import reduxStore from '../../../../../redux-store'
import wrapComponentWithProvider from '../../../../common/provider-wrapped-component'
import RfpActions from '../rfp-actions'
import RfpStatusActions from './actions'
import ReportDefinitionPropType from './report-definition-prop-type'
import RfpStatusRow from './rfp-status-row.jsx'
import RfpStatusSearch from './rfp-status-search.jsx'

export class RfpPlanList extends Component {
  render () {
    return <div style={{ overflowX: 'hidden', overflowY: 'auto' }}>
      <div className='row'>
        <div className='col-md-6' />
        <div className='col-md-6'>
          <RfpStatusSearch />
        </div>
      </div>
      <div className='row'>
        <div className='col-md-12'>
          <table className='table table-sm table-striped'>
            <thead className='thead-light'>
              <tr style={{ textAlign: 'center' }}>
                <th>ID</th>
                <th className='text-left pl-5'>Name</th>
                <th>Created by</th>
                <th>Status</th>
                <th style={{ width: '400px' }}>Reports</th>
              </tr>
            </thead>
            <tbody>
              {
                this.props.isLoadingRfpPlans
                  ? this.renderLoadingIconRow()
                  : this.renderRfpPlanRows()
              }
            </tbody>
          </table>
          {this.renderPagination()}
        </div>
      </div>
    </div>
  }

  renderLoadingIconRow () {
    return <tr>
      <td colSpan={5} className='p-5 text-center'>
        <div className='fa fa-5x fa-spin fa-spinner mb-4' />
        <h4>Loading RFP Plans...</h4>
      </td>
    </tr>
  }

  renderRfpPlanRows () {
    const plansToRender = this.props.rfpPlans.slice(this.props.planListOffset, this.props.planListOffset + this.props.planListLimit)
    return plansToRender.map(rfpPlan => (
      <RfpStatusRow
        key={rfpPlan.id}
        planId={rfpPlan.id}
        name={rfpPlan.name}
        createdById={rfpPlan.createdBy}
        status={rfpPlan.planState}
        reportDefinitions={this.props.rfpReportDefinitions}
      />
    ))
  }

  renderPagination () {
    const numPlans = this.props.rfpPlans.length
    if (numPlans === 0) {
      return null
    }
    const activePageNumber = Math.round(this.props.planListOffset / this.props.planListLimit) + 1
    const numPages = Math.ceil(numPlans / this.props.planListLimit)
    const NUM_PAGES_TO_SHOW = 4 // Number of pages to show before/after the active page
    const startPage = Math.max(activePageNumber - NUM_PAGES_TO_SHOW, 1)
    const endPage = Math.min(activePageNumber + NUM_PAGES_TO_SHOW, numPages)
    const planListLimit = this.props.planListLimit
    const offsetOnClick = pageToLoad => (pageToLoad - 1) * planListLimit
    var pageBlocks = []
    if (activePageNumber > 1) {
      pageBlocks.push(<li key={'Previous'} className='page-item'>
        <a id='rfpPagePrev' className='page-link' href='#' onClick={event => this.props.setPlanListOffset(offsetOnClick(activePageNumber - 1))}>Prev</a>
      </li>)
    }
    for (var iPage = startPage; iPage <= endPage; ++iPage) {
      const liClassName = 'page-item' + (iPage === activePageNumber ? ' active' : '')
      const copyIPage = iPage
      pageBlocks.push(<li key={iPage} className={liClassName}>
        <a id={`rfpPage_${iPage}`} className='page-link' href='#' onClick={event => this.props.setPlanListOffset(offsetOnClick(copyIPage))}>{iPage}</a>
      </li>)
    }
    if (activePageNumber !== numPages) {
      pageBlocks.push(<li key={'Next'} className='page-item'>
        <a id='rfpPageNext' className='page-link' href='#' onClick={event => this.props.setPlanListOffset(offsetOnClick(activePageNumber + 1))}>Next</a>
      </li>)
    }
    return <nav aria-label='RFP plan pagination'>
      <ul className='pagination justify-content-center'>
        {pageBlocks}
      </ul>
    </nav>
  }

  componentDidMount () {
    this.props.loadRfpPlans(this.props.userId)
  }

  componentWillUnmount () {
    this.props.clearRfpPlans()
  }
}

RfpPlanList.propTypes = {
  rfpPlans: PropTypes.array,
  rfpReportDefinitions: ReportDefinitionPropType,
  isLoadingRfpPlans: PropTypes.bool,
  planListOffset: PropTypes.number,
  planListLimit: PropTypes.number,
  userId: PropTypes.number
}

const mapStateToProps = state => ({
  rfpPlans: state.optimization.rfp.rfpPlans,
  rfpReportDefinitions: state.optimization.rfp.rfpReportDefinitions,
  isLoadingRfpPlans: state.optimization.rfp.isLoadingRfpPlans,
  planListOffset: state.optimization.rfp.planListOffset,
  planListLimit: state.optimization.rfp.planListLimit,
  userId: state.user.loggedInUser && state.user.loggedInUser.id
})

const mapDispatchToProps = dispatch => ({
  clearRfpPlans: () => dispatch(RfpStatusActions.clearRfpPlans()),
  loadRfpPlans: (userId) => dispatch(RfpActions.loadRfpPlans(userId)),
  setPlanListOffset: planListOffset => dispatch(RfpStatusActions.setPlanListOffset(planListOffset))
})

const RfpPlanListComponent = wrapComponentWithProvider(reduxStore, RfpPlanList, mapStateToProps, mapDispatchToProps)
export default RfpPlanListComponent
