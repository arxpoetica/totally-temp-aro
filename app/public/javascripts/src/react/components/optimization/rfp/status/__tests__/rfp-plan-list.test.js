/* global jest test expect */
import React from 'react'
import { shallow } from 'enzyme'
import { RfpPlanList } from '../rfp-plan-list.jsx'

// Create a bunch of RFP plans so we can test the pagination
var rfpPlans = []
for (var iPlan = 0; iPlan < 30; ++iPlan) {
  rfpPlans.push({
    id: iPlan,
    name: `RFP Plan ${iPlan}`,
    createdBy: 4
  })
}

// -----------------------------------------------------------------------------
test('Component initialization and teardown', () => {
  const mockLoadRfpPlans = jest.fn()
  const mockClearRfpPlans = jest.fn()
  const component = shallow(
    <RfpPlanList
      rfpPlans={[]}
      userId={42}
      loadRfpPlans={mockLoadRfpPlans}
      clearRfpPlans={mockClearRfpPlans}
    />
  )
  expect(mockLoadRfpPlans).toHaveBeenCalledWith(42)

  component.unmount()
  expect(mockClearRfpPlans).toHaveBeenCalled()
})

// -----------------------------------------------------------------------------
test('Component pagination render', () => {
  const mockLoadRfpPlans = jest.fn()
  const component = shallow(
    <RfpPlanList
      rfpPlans={rfpPlans}
      userId={42}
      planListOffset={0}
      planListLimit={10}
      loadRfpPlans={mockLoadRfpPlans}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('Component pagination navigation', () => {
  const mockLoadRfpPlans = jest.fn()
  const mockSetPlanListOffset = jest.fn()
  const PLANS_PER_PAGE = 10
  const component = shallow(
    <RfpPlanList
      rfpPlans={rfpPlans}
      userId={42}
      planListOffset={0}
      planListLimit={PLANS_PER_PAGE}
      loadRfpPlans={mockLoadRfpPlans}
      setPlanListOffset={mockSetPlanListOffset}
    />
  )
  // We are at page 1
  expect(component).toMatchSnapshot()
  component.find('#rfpPageNext').simulate('click')
  // We are at page 2
  expect(mockSetPlanListOffset).toHaveBeenCalledWith(1 * PLANS_PER_PAGE)
  component.setProps({ planListOffset: PLANS_PER_PAGE })
  expect(component).toMatchSnapshot()
  component.find('#rfpPagePrev').simulate('click')
  // We are at page 1
  expect(mockSetPlanListOffset).toHaveBeenCalledWith(0 * PLANS_PER_PAGE)
  component.setProps({ planListOffset: 0 })

  // Click on page 2
  expect(component).toMatchSnapshot()
  component.find('#rfpPage_2').simulate('click')
  component.setProps({ planListOffset: PLANS_PER_PAGE })
  expect(mockSetPlanListOffset).toHaveBeenCalledWith(1 * PLANS_PER_PAGE)

  // Click on page 3
  expect(component).toMatchSnapshot()
  component.find('#rfpPage_3').simulate('click')
  component.setProps({ planListOffset: 2 * PLANS_PER_PAGE })
  expect(mockSetPlanListOffset).toHaveBeenCalledWith(2 * PLANS_PER_PAGE)

  expect(component).toMatchSnapshot()
})
