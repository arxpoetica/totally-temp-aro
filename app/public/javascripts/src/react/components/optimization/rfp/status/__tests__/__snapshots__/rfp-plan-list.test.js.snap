// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component pagination navigation 1`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="0"
            name="RFP Plan 0"
            planId={0}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="1"
            name="RFP Plan 1"
            planId={1}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="2"
            name="RFP Plan 2"
            planId={2}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="3"
            name="RFP Plan 3"
            planId={3}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="4"
            name="RFP Plan 4"
            planId={4}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="5"
            name="RFP Plan 5"
            planId={5}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="6"
            name="RFP Plan 6"
            planId={6}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="7"
            name="RFP Plan 7"
            planId={7}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="8"
            name="RFP Plan 8"
            planId={8}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="9"
            name="RFP Plan 9"
            planId={9}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item active"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
            key="Next"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPageNext"
              onClick={[Function]}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Component pagination navigation 2`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="10"
            name="RFP Plan 10"
            planId={10}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="11"
            name="RFP Plan 11"
            planId={11}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="12"
            name="RFP Plan 12"
            planId={12}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="13"
            name="RFP Plan 13"
            planId={13}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="14"
            name="RFP Plan 14"
            planId={14}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="15"
            name="RFP Plan 15"
            planId={15}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="16"
            name="RFP Plan 16"
            planId={16}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="17"
            name="RFP Plan 17"
            planId={17}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="18"
            name="RFP Plan 18"
            planId={18}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="19"
            name="RFP Plan 19"
            planId={19}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item"
            key="Previous"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPagePrev"
              onClick={[Function]}
            >
              Prev
            </a>
          </li>
          <li
            className="page-item"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item active"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
            key="Next"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPageNext"
              onClick={[Function]}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Component pagination navigation 3`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="0"
            name="RFP Plan 0"
            planId={0}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="1"
            name="RFP Plan 1"
            planId={1}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="2"
            name="RFP Plan 2"
            planId={2}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="3"
            name="RFP Plan 3"
            planId={3}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="4"
            name="RFP Plan 4"
            planId={4}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="5"
            name="RFP Plan 5"
            planId={5}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="6"
            name="RFP Plan 6"
            planId={6}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="7"
            name="RFP Plan 7"
            planId={7}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="8"
            name="RFP Plan 8"
            planId={8}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="9"
            name="RFP Plan 9"
            planId={9}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item active"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
            key="Next"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPageNext"
              onClick={[Function]}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Component pagination navigation 4`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="10"
            name="RFP Plan 10"
            planId={10}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="11"
            name="RFP Plan 11"
            planId={11}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="12"
            name="RFP Plan 12"
            planId={12}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="13"
            name="RFP Plan 13"
            planId={13}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="14"
            name="RFP Plan 14"
            planId={14}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="15"
            name="RFP Plan 15"
            planId={15}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="16"
            name="RFP Plan 16"
            planId={16}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="17"
            name="RFP Plan 17"
            planId={17}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="18"
            name="RFP Plan 18"
            planId={18}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="19"
            name="RFP Plan 19"
            planId={19}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item"
            key="Previous"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPagePrev"
              onClick={[Function]}
            >
              Prev
            </a>
          </li>
          <li
            className="page-item"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item active"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
            key="Next"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPageNext"
              onClick={[Function]}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Component pagination navigation 5`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="20"
            name="RFP Plan 20"
            planId={20}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="21"
            name="RFP Plan 21"
            planId={21}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="22"
            name="RFP Plan 22"
            planId={22}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="23"
            name="RFP Plan 23"
            planId={23}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="24"
            name="RFP Plan 24"
            planId={24}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="25"
            name="RFP Plan 25"
            planId={25}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="26"
            name="RFP Plan 26"
            planId={26}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="27"
            name="RFP Plan 27"
            planId={27}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="28"
            name="RFP Plan 28"
            planId={28}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="29"
            name="RFP Plan 29"
            planId={29}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item"
            key="Previous"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPagePrev"
              onClick={[Function]}
            >
              Prev
            </a>
          </li>
          <li
            className="page-item"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item active"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Component pagination render 1`] = `
<div
  style={
    Object {
      "overflowX": "hidden",
      "overflowY": "auto",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    />
    <div
      className="col-md-6"
    >
      <Connect(RfpStatusSearch) />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <table
        className="table table-sm table-striped"
      >
        <thead
          className="thead-light"
        >
          <tr
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <th>
              ID
            </th>
            <th
              className="text-left pl-5"
            >
              Name
            </th>
            <th>
              Created by
            </th>
            <th>
              Status
            </th>
            <th
              style={
                Object {
                  "width": "400px",
                }
              }
            >
              Reports
            </th>
          </tr>
        </thead>
        <tbody>
          <Connect(RfpStatusRow)
            createdById={4}
            key="0"
            name="RFP Plan 0"
            planId={0}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="1"
            name="RFP Plan 1"
            planId={1}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="2"
            name="RFP Plan 2"
            planId={2}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="3"
            name="RFP Plan 3"
            planId={3}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="4"
            name="RFP Plan 4"
            planId={4}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="5"
            name="RFP Plan 5"
            planId={5}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="6"
            name="RFP Plan 6"
            planId={6}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="7"
            name="RFP Plan 7"
            planId={7}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="8"
            name="RFP Plan 8"
            planId={8}
          />
          <Connect(RfpStatusRow)
            createdById={4}
            key="9"
            name="RFP Plan 9"
            planId={9}
          />
        </tbody>
      </table>
      <nav
        aria-label="RFP plan pagination"
      >
        <ul
          className="pagination justify-content-center"
        >
          <li
            className="page-item active"
            key="1"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_1"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className="page-item"
            key="2"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_2"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className="page-item"
            key="3"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPage_3"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className="page-item"
            key="Next"
          >
            <a
              className="page-link"
              href="#"
              id="rfpPageNext"
              onClick={[Function]}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
